src/resolve.rs

```diff
@@ -95,6 +98,28 @@ impl CrateDerivation {
             .chain(metadata.workspace_members.iter())
             .any(|pkg_id| *pkg_id == package.id);
 
+        let crate_type = package
+            .targets
+            .iter()
+            .fold(Vec::<String>::new(), |types_vec, target| {
```

Comment:
you probably want a `filter/collect` combination instead.

Reply:
I will look into filter and collect.

For more info take a look at my associated issue: https://github.com/kolloch/crate2nix/issues/31